src/applications/disability-benefits/2346/containers/ConfirmationPage.jsx

```diff
@@ -100,34 +95,50 @@ const ConfirmationPage = ({
 );
 
 ConfirmationPage.propTypes = {
-  email: PropTypes.string.isRequired,
-  confirmationNumber: PropTypes.string.isRequired,
-  shippingAddress: PropTypes.object.isRequired,
+  confirmationPageData: PropTypes.shape({
+    fullName: PropTypes.shape({
+      first: PropTypes.string.isRequired,
+      middle: PropTypes.string,
+      last: PropTypes.string.isRequired,
+    }),
+    email: PropTypes.string.isRequired,
+    order: PropTypes.array.isRequired,
+    shippingAddress: PropTypes.object.isRequired,
+    supplies: PropTypes.array.isRequired,
+  }),
 };
 
 ConfirmationPage.defaultProps = {
-  email: '',
-  shippingAddress: {},
+  confirmationPageData: {
+    fullName: {
+      first: '',
+      last: '',
+    },
+    email: '',
+    order: [],
+    shippingAddress: {},
+    supplies: [],
+  },
 };
 
 const mapStateToProps = state => {
-  const supplies = state.form?.data?.supplies;
-  const selectedProducts = state.form?.data?.selectedProducts;
-  const productIdArray = selectedProducts?.map(product => product.productId);
+  const selectedAddress = state.form?.data['view:currentAddress'];
+  const shippingAddress = state.form?.data[selectedAddress];
+  const { fullName, vetEmail, order, supplies } = state.form?.data;
+  const productIdArray = order?.map(product => product.productId);
   const selectedProductArray = supplies?.filter(supply =>
     productIdArray?.includes(supply.productId),
   );
+
   // Temporary fallback until this is added to the API response
   const submittedAt = state.form?.submission?.submittedAt || moment();
 
   return {
     submittedAt,
-    email: state.form?.data?.email,
```

Comment:
Where is `vetEmail` coming from? Just a passed in prop?

Reply:
Its in state - state.form.data